app/models/admin/collection.rb

```diff
@@ -230,7 +230,10 @@ def create_dropbox_directory!
       
       unless File.directory?(absolute_path)
         begin
-          Dir.mkdir(absolute_path)
+          old_umask = File::umask
+          File::umask(0002)
```

Comment:
Can I suggest adding a comment explaining why you're caching the old umask, changing it for one call, and then restoring it? I'll be lost and have to reconstruct your motivation from guesswork, if I ever have to tweak this code.

Also it might be useful to leave symbolic versions of the octal permissions in the comments. 0775 is rwxrwxr-x, for those playing the home game.